### DISCUSSION TITLE
Firebase SDK  9.0.0: Undefined symbol in zip distribution

### DISCUSSION THREAD

**Discussion by aidiwithV** [firebase/firebase-ios-sdk#9794] [Q&A] [answered]
Sorry   I updated firebase SDK of iOS to 9.0.0 ，xcode13.3.1
I meet an error  
Undefined symbol: _OBJC_CLASS_$_FIRHeartbeatControlle

I can not find this controller
![error](https://user-images.githubusercontent.com/37919448/167984609-15ff3b71-1dd4-433e-9c08-033cf1454651.jpg)

**Comment by ncooke3** [comment, marked as answer]
Hi @aidiwithV,

Sorry for the trouble. Could you share additional details like are you using CocoaPods/SPM/Zip/Carthage? 

Also, if you have a simple sample project that reproduces the issue– that would be very helpful.

**Reply by ncooke3** [reply]
See below thread– the solution was to drag in the `FirebaseCoreInternal.xcframework` from the zip artifact.

**Comment by aidiwithV** [comment]
I used ZIP and these framework
![framework](https://user-images.githubusercontent.com/37919448/167985960-dab470a0-5d87-4cd9-a42e-80996b7482f4.jpg)

**Reply by ncooke3** [reply]
Gotcha, thanks. There is a new `FirebaseCoreInternal.framework` that needs to be dragged in as well. Do you see it in the zip artifact?

**Reply by ncooke3** [reply]
@aidiwithV, I just checked and there should be a `FirebaseCoreInternal.xcframework` in the `Firebase/FirebaseAnalytics` directory of the [zip artifact:](https://github.com/firebase/firebase-ios-sdk/releases/tag/9.0.0) 
<img width="500" alt="image" src="https://user-images.githubusercontent.com/36927374/167986571-34b777e1-e518-44d8-87f0-c673ac061566.png">

I suspect that dragging the `FirebaseCoreInternal.xcframework` under the `FirebaseAnalytics` group should resolve the build issue.

**Reply by aidiwithV** [reply]
I found this   Thank you
Success

**Reply by ncooke3** [reply]
Wonderful! 🚀
